src/modules/BootStorage.rb

```diff
@@ -165,14 +165,24 @@ def bootloader_installable?
       true
     end
 
-    # Find the blkdevice for the filesystem mounted at mountpoint. Returns nil
-    # if no filesystem is found or the filesystem has no blkdevice (e.g. NFS).
+    # Find the partition or disk for the filesystem mounted at mountpoint.
+    #
+    # Returns nil if no filesystem is found or the filesystem has no blkdevice
+    # (e.g. NFS).
+    #
+    # If the filesystem is in a virtual device (like a LUKS or LVM volume), it
+    # returns the (first) underlying partition or disk.
     def find_blk_device_at_mountpoint(mountpoint)
-      fses = Storage::Filesystem.find_by_mountpoint(staging, mountpoint)
+      fses = staging.filesystems.with_mountpoint(mountpoint)
       return nil if fses.empty?
-      return nil if fses[0].blk_devices.empty?
 
-      fses[0].blk_devices[0]
+      partitions = fses.partitions
+      partitions = fses.lvm_vgs.partitions if partitions.empty?
```

Comment:
hmm, I expect that it always return partitions and not need this special lvm handling, if we in future support more such stuff, how complex can be this combination? like lvm, device_mapper, various raids and so on. We already in bootloader fighting with it that we have problem to get physical device for such virtual ones and I hope new storage abstract it.